### PR TITLE
Hotfix/deployment with errors

### PR DESCRIPTION
**Pull request by arbulu89**
Fix the next errors:
- Incorrect usage of ruby's `case` statement. The `||` was not working properly. This was causing the progress data being sent after finished with `0` as percentage
- Display properly the bar styles when the deployment is finished. It was not removing the animated styles
- Set the failed icon in some task if the deployment is finished and it doesn't have 100 as percentage. This can happen as terraform doesn't sent the whole output when it fails...

### CODE REVIEW

**Comment by arbulu89** [comment]
@stefanotorresi @MalloZup Could you have a look on this?

**Review by MalloZup — APPROVED** [review]
looks mergable thx! even if I'm not familiar with the code

File: app/assets/javascripts/deploys.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ function update_tasks(progress_data) {
     progress_text = task_data.progress + "% - " + task_data.text;
     $span_task_id.html(progress_text);
 
-    if (!task_data.success) {
+    if (!task_data.success || (task_data.progress < 100 && finished)) {
```
Comment (MalloZup) [inline app/assets/javascripts/deploys.js:68]: super minor`nit` substitute magic number 100 with constant `const BAR_PERCENTAGE_COMPLETE` or something related

but it still ok even

File: app/assets/javascripts/deploys.js [hunk 2]
```diff
@@ -88,6 +88,18 @@ function update_tasks(progress_data) {
...
+function update_progress(data, finished) {
+  if ("progress" in data) {
+    if ("total_progress" in data.progress) {
+      update_progress_bar(
+          data.progress.total_progress, data.error, finished)
+    }
+    if ("tasks_progress" in data.progress) {
+      update_tasks(data.progress.tasks_progress, finished)
+    }
+  }
```
Comment (stefanotorresi) [inline app/assets/javascripts/deploys.js]: Early returns for the win ;)
```suggestion
  if (!("progress" in data)) {
    return
  }
  if ("total_progress" in data.progress) {
    update_progress_bar(data.progress.total_progress, data.error, finished)
  }
  if ("tasks_progress" in data.progress) {
    update_tasks(data.progress.tasks_progress, finished)
  }
```

**Review by stefanotorresi — APPROVED** [review]
LGTM other than minor refactoring suggested.

File: app/assets/javascripts/deploys.js (hunk 2 above)
Comment (arbulu89) [reply app/assets/javascripts/deploys.js]: Thanks! Changed in https://github.com/SUSE/blue-horizon/pull/24/commits/cd341b4e7fbc3670796a91c18b743238fd3ae4f5

**Pull request merged**